src/main/java/org/jenkinsci/remoting/engine/JnlpProtocol.java

```diff
@@ -49,16 +57,44 @@ protected JnlpProtocol(String secretKey, String slaveName) {
     public abstract String getName();
 
     /**
+     * Performs a handshake with the master and establishes a {@link Channel}.
+     *
+     * @param socket The established {@link Socket} connection to the master.
+     * @param channelBuilder The {@link ChannelBuilder} to use.
+     * @return The established channel if successful, else null.
+     * @throws IOException
+     */
+    @Nullable
+    public Channel establishChannel(Socket socket, ChannelBuilder channelBuilder) throws IOException {
+        DataOutputStream outputStream = new DataOutputStream(socket.getOutputStream());
+        BufferedInputStream inputStream = new BufferedInputStream(socket.getInputStream());
+        if(performHandshake(outputStream, inputStream)) {
+            return buildChannel(socket, channelBuilder);
```

Comment:
Where `0A6C` is the chunk length and `ACED` is the start of the `ObjectOutputStream` in that chunk